lib/blocktree/blocktree_test.go

```diff
@@ -458,38 +453,15 @@ func TestBlockTree_Prune(t *testing.T) {
 	}
 }
 
-func TestBlockTree_PruneCache(t *testing.T) {
-	var bt *BlockTree
-	var branches []testBranch
-
-	for {
-		bt, branches = createTestBlockTree(t, testHeader, 5)
-		if len(branches) > 0 && len(bt.getNode(branches[0].hash).children) > 1 {
-			break
-		}
-	}
-
-	// pick some block to finalise
-	finalised := bt.root.children[0].children[0].children[0]
-	pruned := bt.Prune(finalised.hash)
-
-	for _, prunedHash := range pruned {
-		block, ok := bt.nodeCache[prunedHash]
-
-		require.False(t, ok)
-		require.Nil(t, block)
-	}
-}
-
 func TestBlockTree_GetHashByNumber(t *testing.T) {
 	bt, _ := createTestBlockTree(t, testHeader, 8)
 	best := bt.DeepestBlockHash()
-	bn := bt.nodeCache[best]
+	bn := bt.getNode(best)
 
 	for i := int64(0); i < bn.number.Int64(); i++ {
 		hash, err := bt.GetHashByNumber(big.NewInt(i))
 		require.NoError(t, err)
-		require.Equal(t, big.NewInt(i), bt.nodeCache[hash].number)
```

Comment:
So was the overhead of maintaining the cache just not worth it in this situation? I suppose this means that lookup times in the tree will be quick enough that its better to just lookup than have to maintain the cache?

Reply:
yeah I just don't think it was necessary, the blocktree shouldn't ever grow too deep as we finalize blocks relatively quickly